src/umm_malloc.c

```diff
@@ -278,6 +278,21 @@ static uint16_t umm_assimilate_down(umm_heap *heap, uint16_t c, uint16_t freemas
 /* ------------------------------------------------------------------------- */
 
 void umm_multi_init_heap(umm_heap *heap, void *ptr, size_t size) {
+    /* Guard against too many blocks for 15-bit indices */
+    if ((size / UMM_BLOCKSIZE) > UMM_BLOCKNO_MASK)
```

Comment:
The UMM_BLOCKNO_MASK just happens to be the maximum number of blocks, but I would prefer a separate constant like UMM_MAX_BLOCKNO to clarify the intent. Then we need a compile time check to make sure UMM_MAX_BLOCKNO <= UMM_BLOCKNO_MASK

I would also like to see a test case implemented for this if it's not too much trouble - I use the test descriptions as a spec :-)